File: BrainTranning/app/src/main/java/com/hfad/braintranning/MainActivity.java
package com.hfad.braintranning;

import androidx.appcompat.app.AppCompatActivity;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    private SharedPreferences pref;
    private EditText editSave;
    private final String save_key = "save_key";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        init();
    }

    public void onClickSave(View view) {
        SharedPreferences.Editor edit = pref.edit();
        edit.putString(save_key,editSave.getText().toString());
        edit.apply();
    }

    public void onClickGet(View view) {
        editSave.setText(pref.getString(save_key, "NULL"));
    }
    private void init() {
        pref = getSharedPreferences("Test", MODE_PRIVATE);
        editSave = findViewById(R.id.editTextSave);
        editSave.setText(pref.getString(save_key, "NULL"));
    }
}

File: LesTwoSome/app/src/main/java/com/hfad/lestwosome/MainActivity.kt
package com.hfad.lestwosome

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

    }

    fun onClickStart(view: View) {
        var intent = Intent(this, SecondActivity::class.java).apply {
            putExtra(
                "some",
                "That shit easy for me!"
            )
        }
        startActivity(intent)
    }

}